This assignment explores the venues in the city of Toronto with the postal code beginning with M.

Project Title: **Which location to open a health food store?**

**Introduction**: As people are paying increasing attention to their health. There is an increasing demand in health food,whole food and superfood. Finding the proper location to open a health food store is important for business success. This assignment help to solve this problem by using Foursquare venue data in the city of Toronto. 
Each neighborhood will be explored. The top 3 frequent venues will be selected and analyzed to find out which location is proper to open a health food store.

**Data**:
City of Toronto borough, neighborhood, postal codes were collected from "https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M".
Venues data of Toronto was collected using Foursquare.
Toronto neighborhoods population size were collected from "https://en.wikipedia.org/wiki/Demographics_of_Toronto_neighbourhoods"

In [1]:
import numpy as np 
import pandas as pd 
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

import json 

!conda install -c conda-forge geopy --yes 
from geopy.geocoders import Nominatim 

import requests 
from pandas.io.json import json_normalize 

import matplotlib.cm as cm
import matplotlib.colors as colors

from sklearn.cluster import KMeans

!conda install -c conda-forge folium=0.5.0 --yes 
import folium

Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 4.5.11
  latest version: 4.8.1

Please update conda by running

    $ conda update -n base -c defaults conda



## Package Plan ##

  environment location: /home/jupyterlab/conda/envs/python

  added / updated specs: 
    - geopy


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    certifi-2019.11.28         |           py36_0         149 KB  conda-forge
    scikit-learn-0.20.1        |   py36h22eb022_0         5.7 MB
    liblapack-3.8.0            |      11_openblas          10 KB  conda-forge
    liblapacke-3.8.0           |      11_openblas          10 KB  conda-forge
    geographiclib-1.50         |             py_0          34 KB  conda-forge
    libopenblas-0.3.6          |       h5a2b251_2         7.7 MB
    scipy-1.4.1                |   py36h921218d_0        18.9 MB  conda-forge
    lib

In [2]:
df=pd.read_csv('week3 longitude and altitude.csv')

In [3]:
borough=pd.value_counts(df['BOROUGH'])
borough

North York          24
Downtown Toronto    19
Scarborough         17
Etobicoke           11
Central Toronto      9
West Toronto         6
York                 5
East York            5
East Toronto         5
Mississauga          1
Queen's Park         1
Name: BOROUGH, dtype: int64

In [4]:
borough.shape

(11,)

In [5]:
neighborhood=pd.value_counts(df['NEIGHBORHOOD'])
neighborhood.shape

(102,)

There are 11 boroughs and 102 neighborhoods in the city of Toronto.

To create a map of city of Toronto including 11 boroughs

In [6]:
latitude = 43.6532
longitude= -79.3832
map_toronto= folium.Map(location=[latitude, longitude], zoom_start=11)
for lat, lng, label in zip(df['Latitude'], df['Longitude'], df['NEIGHBORHOOD']):
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=3,
        popup=label,
        color='',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_toronto)  
    
map_toronto

In [7]:
CLIENT_ID = 'N043PBNGAO2F3G1AZ1UB1ICLOCFZMGCXO0RYQTA0WMNUBFNU' 
CLIENT_SECRET = 'SX4CB1W2ZDDX3SIEHUJZJYYBM5XJJP4412QUZTKDNC1VPMBK' 
VERSION = '20180605' 

In [8]:
LIMIT = 100 
radius = 500
def getNearbyVenues(names, latitudes, longitudes, radius=500):
    
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
    
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
            
       
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Neighborhood', 
                  'Neighborhood Latitude', 
                  'Neighborhood Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    
    return(nearby_venues)

In [9]:
venues = getNearbyVenues(names=df['NEIGHBORHOOD'],
                         latitudes=df['Latitude'],
                         longitudes=df['Longitude'])

Rouge, Malvern
Highland Creek, Rouge Hill, Port Union
Guildwood, Morningside, West Hill
Woburn
Cedarbrae
Scarborough Village
East Birchmount Park, Ionview, Kennedy Park
Clairlea, Golden Mile, Oakridge
Cliffcrest, Cliffside, Scarborough Village West
Birch Cliff, Cliffside West
Dorset Park, Scarborough Town Centre, Wexford Heights
Maryvale, Wexford
Agincourt
Clarks Corners, Sullivan, Tam O'Shanter
Agincourt North, L'Amoreaux East, Milliken, Steeles East
L'Amoreaux West
Upper Rouge
Hillcrest Village
Fairview, Henry Farm, Oriole
Bayview Village
Silver Hills, York Mills
Newtonbrook, Willowdale
Willowdale South
York Mills West
Willowdale West
Parkwoods
Don Mills North
Flemingdon Park, Don Mills South
Bathurst Manor, Downsview North, Wilson Heights
Northwood Park, York University
CFB Toronto, Downsview East
Downsview West
Downsview Central
Downsview Northwest
Victoria Village
Woodbine Gardens, Parkview Hill
Woodbine Heights
The Beaches
Leaside
Thorncliffe Park
East Toronto
The Danforth West, 

In [10]:
venues.groupby('Neighborhood').count()

Neighborhood Latitude  \
Neighborhood                                                                
Adelaide, King, Richmond                                              100   
Agincourt                                                               4   
Agincourt North, L'Amoreaux East, Milliken, Ste...                      2   
Albion Gardens, Beaumond Heights, Humbergate, J...                     11   
Alderwood, Long Branch                                                  9   
Bathurst Manor, Downsview North, Wilson Heights                        20   
Bayview Village                                                         4   
Bedford Park, Lawrence Manor East                                      23   
Berczy Park                                                            56   
Birch Cliff, Cliffside West                                             4   
Bloordale Gardens, Eringate, Markland Wood, Old...                      7   
Brockton, Exhibition Place, Parkdale Village                           22   
Business Reply Mail Processing Centre 969 Eastern                      17   
CFB Toronto, Downsview East                                             5   
CN Tower, Bathurst Quay, Island airport, Harbou...                     16   
Cabbagetown, St. James Town                                            46   
Caledonia-Fairbanks                                                     5   
Canada Post Gateway Processing Centre                                  11   
Cedarbrae                                                               8   
Central Bay Street                                                     86   
Chinatown, Grange Park, Kensington Market                              85   
Christie                                                               18   
Church and Wellesley                                                   84   
Clairlea, Golden Mile, Oakridge                                        10   
Clarks Corners, Sullivan, Tam O'Shanter                                13   
Cliffcrest, Cliffside, Scarborough Village West                         2   
Commerce Court, Victoria Hotel                                        100   
Davisville                                                             31   
Davisville North                                                        7   
Deer Park, Forest Hill SE, Rathnelly, South Hil...                     14   
Del Ray, Keelesdale, Mount Dennis, Silverthorn                          6   
Design Exchange, Toronto Dominion Centre                              100   
Don Mills North                                                         4   
Dorset Park, Scarborough Town Centre, Wexford H...                      6   
Dovercourt Village, Dufferin                                           14   
Downsview Central                                                       3   
Downsview Northwest                                                     5   
Downsview West                                                          5   
Downsview, North Park, Upwood Park                                      4   
East Birchmount Park, Ionview, Kennedy Park                             6   
East Toronto                                                            3   
Emery, Humberlea                                                        1   
Fairview, Henry Farm, Oriole                                           62   
First Canadian Place, Underground city                                100   
Flemingdon Park, Don Mills South                                       23   
Forest Hill North, Forest Hill West                                     4   
Glencairn                                                               4   
Guildwood, Morningside, West Hill                                       7   
Harbord, University of Toronto                                         36   
Harbourfront                                                           48   
Harbourfront East, Toronto Islands, Union Statio

In [11]:
print('There are {} uniques categories.'.format(len(venues['Venue Category'].unique())))

There are 268 uniques categories.


Analyze each neighborhood in the city of Toronto

In [12]:
toronto_onehot = pd.get_dummies(venues[['Venue Category']], prefix="", prefix_sep="")

toronto_onehot['Neighborhood'] = venues['Neighborhood'] 

fixed_columns = [toronto_onehot.columns[-1]] + list(toronto_onehot.columns[:-1])
toronto_onehot = toronto_onehot[fixed_columns]

In [13]:
toronto_onehot.shape

(2215, 268)

calculate the frequency of each categories of venue

In [14]:
toronto_grouped = toronto_onehot.groupby('Neighborhood').mean().reset_index()
toronto_grouped

Neighborhood  Yoga Studio  \
0                            Adelaide, King, Richmond     0.000000   
1                                           Agincourt     0.000000   
2   Agincourt North, L'Amoreaux East, Milliken, St...     0.000000   
3   Albion Gardens, Beaumond Heights, Humbergate, ...     0.000000   
4                              Alderwood, Long Branch     0.000000   
5     Bathurst Manor, Downsview North, Wilson Heights     0.000000   
6                                     Bayview Village     0.000000   
7                   Bedford Park, Lawrence Manor East     0.000000   
8                                         Berczy Park     0.000000   
9                         Birch Cliff, Cliffside West     0.000000   
10  Bloordale Gardens, Eringate, Markland Wood, Ol...     0.000000   
11       Brockton, Exhibition Place, Parkdale Village     0.000000   
12  Business Reply Mail Processing Centre 969 Eastern     0.058824   
13                        CFB Toronto, Downsview East     0.000000   
14  CN Tower, Bathurst Quay, Island airport, Harbo...     0.000000   
15                        Cabbagetown, St. James Town     0.000000   
16                                Caledonia-Fairbanks     0.000000   
17              Canada Post Gateway Processing Centre     0.000000   
18                                          Cedarbrae     0.000000   
19                                 Central Bay Street     0.011628   
20          Chinatown, Grange Park, Kensington Market     0.000000   
21                                           Christie     0.000000   
22                               Church and Wellesley     0.011905   
23                    Clairlea, Golden Mile, Oakridge     0.000000   
24            Clarks Corners, Sullivan, Tam O'Shanter     0.000000   
25    Cliffcrest, Cliffside, Scarborough Village West     0.000000   
26                     Commerce Court, Victoria Hotel     0.000000   
27                                         Davisville     0.000000   
28                                   Davisville North     0.000000   
29  Deer Park, Forest Hill SE, Rathnelly, South Hi...     0.000000   
30     Del Ray, Keelesdale, Mount Dennis, Silverthorn     0.000000   
31           Design Exchange, Toronto Dominion Centre     0.000000   
32                                    Don Mills North     0.000000   
33  Dorset Park, Scarborough Town Centre, Wexford ...     0.000000   
34                       Dovercourt Village, Dufferin     0.000000   
35                                  Downsview Central     0.000000   
36                                Downsview Northwest     0.000000   
37                                     Downsview West     0.000000   
38                 Downsview, North Park, Upwood Park     0.000000   
39        East Birchmount Park, Ionview, Kennedy Park     0.000000   
40                                       East Toronto     0.000000   
41                                   Emery, Humberlea     0.000000   
42                       Fairview, Henry Farm, Oriole     0.016129   
43             First Canadian Place, Underground city     0.000000   
44                   Flemingdon Park, Don Mills South     0.000000   
45                Forest Hill North, Forest Hill West     0.000000   
46                                          Glencairn     0.000000   
47                  Guildwood, Morningside, West Hill     0.000000   
48                     Harbord, University of Toronto     0.000000   
49                                       Harbourfront     0.020833   
50  Harbourfront East, Toronto Islands, Union Station     0.000000   
51                      High Park, The Junction South     0.000000   
52             Highland Creek, Rouge Hill, Port Union     0.000000   
53                                  Hillcrest Village     0.000000   
54       Humber Bay Shores, Mimico South, New Toronto     0.000000   
55  Humber Bay, King's Mill Park, Kingsway Park So...     0.000000   
56                                      Humber Summ

List the top 10 venues in each neighborhood

In [15]:
def return_most_common_venues(row, num_top_venues):
    row_categories = row.iloc[1:]
    row_categories_sorted = row_categories.sort_values(ascending=False)
    
    return row_categories_sorted.index.values[0:num_top_venues]

In [16]:
num_top_venues = 10

indicators = ['st', 'nd', 'rd']

columns = ['Neighborhood']
for ind in np.arange(num_top_venues):
    try:
        columns.append('{}{} Most Common Venue'.format(ind+1, indicators[ind]))
    except:
        columns.append('{}th Most Common Venue'.format(ind+1))

neighborhoods_venues_sorted = pd.DataFrame(columns=columns)
neighborhoods_venues_sorted['Neighborhood'] = toronto_grouped['Neighborhood']

for ind in np.arange(toronto_grouped.shape[0]):
    neighborhoods_venues_sorted.iloc[ind, 1:] = return_most_common_venues(toronto_grouped.iloc[ind, :], num_top_venues)

neighborhoods_venues_sorted

Neighborhood  \
0                            Adelaide, King, Richmond   
1                                           Agincourt   
2   Agincourt North, L'Amoreaux East, Milliken, St...   
3   Albion Gardens, Beaumond Heights, Humbergate, ...   
4                              Alderwood, Long Branch   
5     Bathurst Manor, Downsview North, Wilson Heights   
6                                     Bayview Village   
7                   Bedford Park, Lawrence Manor East   
8                                         Berczy Park   
9                         Birch Cliff, Cliffside West   
10  Bloordale Gardens, Eringate, Markland Wood, Ol...   
11       Brockton, Exhibition Place, Parkdale Village   
12  Business Reply Mail Processing Centre 969 Eastern   
13                        CFB Toronto, Downsview East   
14  CN Tower, Bathurst Quay, Island airport, Harbo...   
15                        Cabbagetown, St. James Town   
16                                Caledonia-Fairbanks   
17              Canada Post Gateway Processing Centre   
18                                          Cedarbrae   
19                                 Central Bay Street   
20          Chinatown, Grange Park, Kensington Market   
21                                           Christie   
22                               Church and Wellesley   
23                    Clairlea, Golden Mile, Oakridge   
24            Clarks Corners, Sullivan, Tam O'Shanter   
25    Cliffcrest, Cliffside, Scarborough Village West   
26                     Commerce Court, Victoria Hotel   
27                                         Davisville   
28                                   Davisville North   
29  Deer Park, Forest Hill SE, Rathnelly, South Hi...   
30     Del Ray, Keelesdale, Mount Dennis, Silverthorn   
31           Design Exchange, Toronto Dominion Centre   
32                                    Don Mills North   
33  Dorset Park, Scarborough Town Centre, Wexford ...   
34                       Dovercourt Village, Dufferin   
35                                  Downsview Central   
36                                Downsview Northwest   
37                                     Downsview West   
38                 Downsview, North Park, Upwood Park   
39        East Birchmount Park, Ionview, Kennedy Park   
40                                       East Toronto   
41                                   Emery, Humberlea   
42                       Fairview, Henry Farm, Oriole   
43             First Canadian Place, Underground city   
44                   Flemingdon Park, Don Mills South   
45                Forest Hill North, Forest Hill West   
46                                          Glencairn   
47                  Guildwood, Morningside, West Hill   
48                     Harbord, University of Toronto   
49                                       Harbourfront   
50  Harbourfront East, Toronto Islands, Union Station   
51                      High Park, The Junction South   
52             Highland Creek, Rouge Hill, Port Union   
53                                  Hillcrest Village   
54       Humber Bay Shores, Mimico South, New Toronto   
55  Humber Bay, King's Mill Park, Kingsway Park So...   
56                                      Humber Summit   
57                                 Humewood-Cedarvale   
58  Kingsview Village, Martin Grove Gardens, Richv...   
59  Kingsway Park South West, Mimico NW, The Queen...   
60                                    L'Amoreaux West   
61                   Lawrence Heights, Lawrence Manor   
62                                      Lawrence Park   
63                                            Leaside   
64                           Little Portugal, Trinity   
65                                  Maryvale, Wexford   
66                        Moore Park, Summerhill East   
67                            Newtonbrook, Willowdale   
68                                 North Toronto West   
69                                          Northwe

In [17]:
kclusters = 5

toronto_grouped_clustering = toronto_grouped.drop('Neighborhood', 1)

kmeans = KMeans(n_clusters=kclusters, random_state=0).fit(toronto_grouped_clustering)

kmeans.labels_[0:10]

array([1, 1, 0, 1, 1, 1, 1, 1, 1, 1], dtype=int32)

In [18]:
toronto_merged =df

toronto_merged =toronto_merged.join(neighborhoods_venues_sorted.set_index('Neighborhood'), on='NEIGHBORHOOD')

toronto_merged

POSTALCODE           BOROUGH  \
0          M1B       Scarborough   
1          M1C       Scarborough   
2          M1E       Scarborough   
3          M1G       Scarborough   
4          M1H       Scarborough   
5          M1J       Scarborough   
6          M1K       Scarborough   
7          M1L       Scarborough   
8          M1M       Scarborough   
9          M1N       Scarborough   
10         M1P       Scarborough   
11         M1R       Scarborough   
12         M1S       Scarborough   
13         M1T       Scarborough   
14         M1V       Scarborough   
15         M1W       Scarborough   
16         M1X       Scarborough   
17         M2H        North York   
18         M2J        North York   
19         M2K        North York   
20         M2L        North York   
21         M2M        North York   
22         M2N        North York   
23         M2P        North York   
24         M2R        North York   
25         M3A        North York   
26         M3B        North York   
27         M3C        North York   
28         M3H        North York   
29         M3J        North York   
30         M3K        North York   
31         M3L        North York   
32         M3M        North York   
33         M3N        North York   
34         M4A        North York   
35         M4B         East York   
36         M4C         East York   
37         M4E      East Toronto   
38         M4G         East York   
39         M4H         East York   
40         M4J         East York   
41         M4K      East Toronto   
42         M4L      East Toronto   
43         M4M      East Toronto   
44         M4N   Central Toronto   
45         M4P   Central Toronto   
46         M4R   Central Toronto   
47         M4S   Central Toronto   
48         M4T   Central Toronto   
49         M4V   Central Toronto   
50         M4W  Downtown Toronto   
51         M4X  Downtown Toronto   
52         M4Y  Downtown Toronto   
53         M5A  Downtown Toronto   
54         M5B  Downtown Toronto   
55         M5C  Downtown Toronto   
56         M5E  Downtown Toronto   
57         M5G  Downtown Toronto   
58         M5H  Downtown Toronto   
59         M5J  Downtown Toronto   
60         M5K  Downtown Toronto   
61         M5L  Downtown Toronto   
62         M5M        North York   
63         M5N   Central Toronto   
64         M5P   Central Toronto   
65         M5R   Central Toronto   
66         M5S  Downtown Toronto   
67         M5T  Downtown Toronto   
68         M5V  Downtown Toronto   
69         M5W  Downtown Toronto   
70         M5X  Downtown Toronto   
71         M6A        North York   
72         M6B        North York   
73         M6C              York   
74         M6E              York   
75         M6G  Downtown Toronto   
76         M6H      West Toronto   
77         M6J      West Toronto   
78         M6K      West Toronto   
79         M6L        North York   
80         M6M              York   
81         M6N              York   
82         M6P      West Toronto   
83         M6R      West Toronto   
84         M6S      West Toronto   
85         M7A  Downtown Toronto   
86         M7R       Mississauga   
87         M7Y      East Toronto   
88         M8V         Etobicoke   
89         M8W         Etobicoke   
90         M8X         Etobicoke   
91         M8Y         Etobicoke   
92         M8Z         Etobicoke   
93         M9A      Queen's Park   
94         M9B         Etobicoke   
95         M9C         Etobicoke   
96         M9L        North York   
97         M9M        North York   
98         M9N              York   
99         M9P         Etobicoke   
100        M9R         Etobicoke   
101        M9V         Etobicoke   
102        M9W         Etobicoke   

                                          NEIGHBORHOOD   Latitude  Longitude  \
0                                       Rouge, Malvern  43.806686 -79.194353   
1               Highland Creek, Rouge Hill, Port Union  43.784535 -79.160497   
2                  

In [19]:
north_york=toronto_merged.loc[toronto_merged['BOROUGH']=='North York']
north_york

POSTALCODE     BOROUGH                                     NEIGHBORHOOD  \
17        M2H  North York                                Hillcrest Village   
18        M2J  North York                     Fairview, Henry Farm, Oriole   
19        M2K  North York                                  Bayview Village   
20        M2L  North York                         Silver Hills, York Mills   
21        M2M  North York                          Newtonbrook, Willowdale   
22        M2N  North York                                 Willowdale South   
23        M2P  North York                                  York Mills West   
24        M2R  North York                                  Willowdale West   
25        M3A  North York                                        Parkwoods   
26        M3B  North York                                  Don Mills North   
27        M3C  North York                 Flemingdon Park, Don Mills South   
28        M3H  North York  Bathurst Manor, Downsview North, Wilson Heights   
29        M3J  North York                  Northwood Park, York University   
30        M3K  North York                      CFB Toronto, Downsview East   
31        M3L  North York                                   Downsview West   
32        M3M  North York                                Downsview Central   
33        M3N  North York                              Downsview Northwest   
34        M4A  North York                                 Victoria Village   
62        M5M  North York                Bedford Park, Lawrence Manor East   
71        M6A  North York                 Lawrence Heights, Lawrence Manor   
72        M6B  North York                                        Glencairn   
79        M6L  North York               Downsview, North Park, Upwood Park   
96        M9L  North York                                    Humber Summit   
97        M9M  North York                                 Emery, Humberlea   

     Latitude  Longitude     1st Most Common Venue      2nd Most Common Venue  \
17  43.803762 -79.363452  Mediterranean Restaurant                Golf Course   
18  43.778517 -79.346556            Clothing Store       Fast Food Restaurant   
19  43.786947 -79.385975        Chinese Restaurant                       Café   
20  43.757490 -79.374714                 Cafeteria              Deli / Bodega   
21  43.789053 -79.408493                 Piano Bar              Deli / Bodega   
22  43.770120 -79.408493          Ramen Restaurant                       Café   
23  43.752758 -79.400049                      Park                       Bank   
24  43.782736 -79.442259                  Pharmacy                Coffee Shop   
25  43.753259 -79.329656                      Park          Food & Drink Shop   
26  43.745906 -79.352188                      Café       Caribbean Restaurant   
27  43.725900 -79.340923                       Gym                Coffee Shop   
28  43.754328 -79.442259               Coffee Shop  Middle Eastern Restaurant   
29  43.767980 -79.487262        Miscellaneous Shop                Coffee Shop   
30  43.737473 -79.464763                      Park                    Airport   
31  43.739015 -79.506944             Grocery Store                       Bank   
32  43.728496 -79.495697                Food Truck               Home Service   
33  43.761631 -79.520999             Grocery Store       Gym / Fitness Center   
34  43.725882 -79.315572               Coffee Shop                Pizza Place   
62  43.733283 -79.419750        Italian Restaurant                Coffee Shop   
71  43.718518 -79.464763            Clothing Store          Accessories Store   
72  43.709577 -79.445073          Asian Restaurant        Japanese Restaurant   
79  43.713756 -79.490074                      Park                     Bakery   
96  43.756303 -79.565963               Pizza Place        Empanada Restaurant   
97  43.724766 -79.532242            Baseball Field              Women's Store   

         3rd Most Common V

In [20]:
north_york=north_york[['BOROUGH','NEIGHBORHOOD','1st Most Common Venue','2nd Most Common Venue','3rd Most Common Venue']]
north_york

BOROUGH                                     NEIGHBORHOOD  \
17  North York                                Hillcrest Village   
18  North York                     Fairview, Henry Farm, Oriole   
19  North York                                  Bayview Village   
20  North York                         Silver Hills, York Mills   
21  North York                          Newtonbrook, Willowdale   
22  North York                                 Willowdale South   
23  North York                                  York Mills West   
24  North York                                  Willowdale West   
25  North York                                        Parkwoods   
26  North York                                  Don Mills North   
27  North York                 Flemingdon Park, Don Mills South   
28  North York  Bathurst Manor, Downsview North, Wilson Heights   
29  North York                  Northwood Park, York University   
30  North York                      CFB Toronto, Downsview East   
31  North York                                   Downsview West   
32  North York                                Downsview Central   
33  North York                              Downsview Northwest   
34  North York                                 Victoria Village   
62  North York                Bedford Park, Lawrence Manor East   
71  North York                 Lawrence Heights, Lawrence Manor   
72  North York                                        Glencairn   
79  North York               Downsview, North Park, Upwood Park   
96  North York                                    Humber Summit   
97  North York                                 Emery, Humberlea   

       1st Most Common Venue      2nd Most Common Venue  \
17  Mediterranean Restaurant                Golf Course   
18            Clothing Store       Fast Food Restaurant   
19        Chinese Restaurant                       Café   
20                 Cafeteria              Deli / Bodega   
21                 Piano Bar              Deli / Bodega   
22          Ramen Restaurant                       Café   
23                      Park                       Bank   
24                  Pharmacy                Coffee Shop   
25                      Park          Food & Drink Shop   
26                      Café       Caribbean Restaurant   
27                       Gym                Coffee Shop   
28               Coffee Shop  Middle Eastern Restaurant   
29        Miscellaneous Shop                Coffee Shop   
30                      Park                    Airport   
31             Grocery Store                       Bank   
32                Food Truck               Home Service   
33             Grocery Store       Gym / Fitness Center   
34               Coffee Shop                Pizza Place   
62        Italian Restaurant                Coffee Shop   
71            Clothing Store          Accessories Store   
72          Asian Restaurant        Japanese Restaurant   
79                      Park                     Bakery   
96               Pizza Place        Empanada Restaurant   
97            Baseball Field              Women's Store   

         3rd Most Common Venue  
17                     Dog Run  
18                 Coffee Shop  
19                        Bank  
20        Ethiopian Restaurant  
21        Ethiopian Restaurant  
22                  Restaurant  
23                         Bar  
24              Discount Store  
25  Construction & Landscaping  
26         Japanese Restaurant  
27            Asian Restaurant  
28          Frozen Yogurt Shop  
29              Massage Studio  
30                  Playground  
31                       Hotel  
32              Baseball Field  
33                Liquor Store  
34                Hockey Arena  
62        Fast Food Restaurant  
71      Furniture / Home Store  
72            Sushi Restaurant  
79  Construction & Landscaping  
96               Women's Store  
97                  Donut Shop

In [21]:
fre1=pd.value_counts(north_york['1st Most Common Venue'])
fre1

Park                        4
Grocery Store               2
Coffee Shop                 2
Clothing Store              2
Asian Restaurant            1
Café                        1
Miscellaneous Shop          1
Baseball Field              1
Piano Bar                   1
Food Truck                  1
Italian Restaurant          1
Cafeteria                   1
Pharmacy                    1
Pizza Place                 1
Ramen Restaurant            1
Chinese Restaurant          1
Mediterranean Restaurant    1
Gym                         1
Name: 1st Most Common Venue, dtype: int64

In [22]:
fre2=pd.value_counts(north_york['2nd Most Common Venue'])
fre2

Coffee Shop                  4
Café                         2
Bank                         2
Deli / Bodega                2
Home Service                 1
Empanada Restaurant          1
Gym / Fitness Center         1
Airport                      1
Caribbean Restaurant         1
Food & Drink Shop            1
Fast Food Restaurant         1
Accessories Store            1
Japanese Restaurant          1
Bakery                       1
Middle Eastern Restaurant    1
Golf Course                  1
Women's Store                1
Pizza Place                  1
Name: 2nd Most Common Venue, dtype: int64

In [23]:
downtown_toronto=toronto_merged.loc[toronto_merged['BOROUGH']=='Downtown Toronto']
downtown_toronto

POSTALCODE           BOROUGH  \
50        M4W  Downtown Toronto   
51        M4X  Downtown Toronto   
52        M4Y  Downtown Toronto   
53        M5A  Downtown Toronto   
54        M5B  Downtown Toronto   
55        M5C  Downtown Toronto   
56        M5E  Downtown Toronto   
57        M5G  Downtown Toronto   
58        M5H  Downtown Toronto   
59        M5J  Downtown Toronto   
60        M5K  Downtown Toronto   
61        M5L  Downtown Toronto   
66        M5S  Downtown Toronto   
67        M5T  Downtown Toronto   
68        M5V  Downtown Toronto   
69        M5W  Downtown Toronto   
70        M5X  Downtown Toronto   
75        M6G  Downtown Toronto   
85        M7A  Downtown Toronto   

                                         NEIGHBORHOOD   Latitude  Longitude  \
50                                           Rosedale  43.679563 -79.377529   
51                        Cabbagetown, St. James Town  43.667967 -79.367675   
52                               Church and Wellesley  43.665860 -79.383160   
53                                       Harbourfront  43.654260 -79.360636   
54                           Ryerson, Garden District  43.657162 -79.378937   
55                                     St. James Town  43.651494 -79.375418   
56                                        Berczy Park  43.644771 -79.373306   
57                                 Central Bay Street  43.657952 -79.387383   
58                           Adelaide, King, Richmond  43.650571 -79.384568   
59  Harbourfront East, Toronto Islands, Union Station  43.640816 -79.381752   
60           Design Exchange, Toronto Dominion Centre  43.647177 -79.381576   
61                     Commerce Court, Victoria Hotel  43.648198 -79.379817   
66                     Harbord, University of Toronto  43.662696 -79.400049   
67          Chinatown, Grange Park, Kensington Market  43.653206 -79.400049   
68  CN Tower, Bathurst Quay, Island airport, Harbo...  43.628947 -79.394420   
69                    Stn A PO Boxes 25 The Esplanade  43.646435 -79.374846   
70             First Canadian Place, Underground city  43.648429 -79.382280   
75                                           Christie  43.669542 -79.422564   
85                                       Queen's Park  43.662301 -79.389494   

   1st Most Common Venue  2nd Most Common Venue 3rd Most Common Venue  \
50                  Park             Playground                 Trail   
51           Coffee Shop            Pizza Place    Italian Restaurant   
52           Coffee Shop    Japanese Restaurant            Restaurant   
53           Coffee Shop                   Park                Bakery   
54           Coffee Shop         Clothing Store                  Café   
55           Coffee Shop                   Café            Restaurant   
56           Coffee Shop           Cocktail Bar        Farmers Market   
57           Coffee Shop                   Café        Ice Cream Shop   
58           Coffee Shop                   Café            Steakhouse   
59           Coffee Shop               Aquarium                 Hotel   
60           Coffee Shop                   Café                 Hotel   
61           Coffee Shop                   Café                 Hotel   
66                  Café    Japanese Restaurant                   Bar   
67                  Café  Vietnamese Restaurant    Chinese Restaurant   
68        Airport Lounge        Airport Service      Airport Terminal   
69           Coffee Shop                   Café    Seafood Restaurant   
70           Coffee Shop                   Café            Steakhouse   
75         Grocery Store                   Café                  Park   
85           Coffee Shop                    Gym                  Park   

   4th Most Common Venue          5th Most Common Venue  \
50               Dog Run               Department Store   
51                  Park                           Café   
52               Gay Bar               Sushi Restaurant   
53                   Pub

In [24]:
fre1=pd.value_counts(downtown_toronto['1st Most Common Venue'])
fre1

Coffee Shop       14
Café               2
Grocery Store      1
Park               1
Airport Lounge     1
Name: 1st Most Common Venue, dtype: int64

In [25]:
fre2=pd.value_counts(downtown_toronto['2nd Most Common Venue'])
fre2

Café                     8
Japanese Restaurant      2
Clothing Store           1
Cocktail Bar             1
Park                     1
Vietnamese Restaurant    1
Airport Service          1
Gym                      1
Pizza Place              1
Playground               1
Aquarium                 1
Name: 2nd Most Common Venue, dtype: int64

In [26]:
scarborough=toronto_merged.loc[toronto_merged['BOROUGH']=='Scarborough']
scarborough

POSTALCODE      BOROUGH                                       NEIGHBORHOOD  \
0         M1B  Scarborough                                     Rouge, Malvern   
1         M1C  Scarborough             Highland Creek, Rouge Hill, Port Union   
2         M1E  Scarborough                  Guildwood, Morningside, West Hill   
3         M1G  Scarborough                                             Woburn   
4         M1H  Scarborough                                          Cedarbrae   
5         M1J  Scarborough                                Scarborough Village   
6         M1K  Scarborough        East Birchmount Park, Ionview, Kennedy Park   
7         M1L  Scarborough                    Clairlea, Golden Mile, Oakridge   
8         M1M  Scarborough    Cliffcrest, Cliffside, Scarborough Village West   
9         M1N  Scarborough                        Birch Cliff, Cliffside West   
10        M1P  Scarborough  Dorset Park, Scarborough Town Centre, Wexford ...   
11        M1R  Scarborough                                  Maryvale, Wexford   
12        M1S  Scarborough                                          Agincourt   
13        M1T  Scarborough            Clarks Corners, Sullivan, Tam O'Shanter   
14        M1V  Scarborough  Agincourt North, L'Amoreaux East, Milliken, St...   
15        M1W  Scarborough                                    L'Amoreaux West   
16        M1X  Scarborough                                        Upper Rouge   

     Latitude  Longitude      1st Most Common Venue 2nd Most Common Venue  \
0   43.806686 -79.194353       Fast Food Restaurant      Department Store   
1   43.784535 -79.160497                        Bar         Women's Store   
2   43.763573 -79.188711               Intersection     Electronics Store   
3   43.770992 -79.216917                Coffee Shop     Korean Restaurant   
4   43.773136 -79.239476                Gas Station                Bakery   
5   43.744734 -79.239476                        Spa     Convenience Store   
6   43.727929 -79.262029           Department Store           Bus Station   
7   43.711112 -79.284577                     Bakery              Bus Line   
8   43.716316 -79.239476        American Restaurant                 Motel   
9   43.692657 -79.264848            College Stadium                  Café   
10  43.757410 -79.273304          Indian Restaurant             Pet Store   
11  43.750071 -79.295849  Middle Eastern Restaurant           Auto Garage   
12  43.794200 -79.262029  Latin American Restaurant                Lounge   
13  43.781638 -79.304302                Pizza Place     Convenience Store   
14  43.815252 -79.284577                       Park            Playground   
15  43.799525 -79.318389       Fast Food Restaurant    Chinese Restaurant   
16  43.836125 -79.205636                        NaN                   NaN   

    3rd Most Common Venue  4th Most Common Venue 5th Most Common Venue  \
0             Event Space   Ethiopian Restaurant   Empanada Restaurant   
1              Donut Shop     Dim Sum Restaurant                 Diner   
2     Rental Car Location            Pizza Place        Breakfast Spot   
3       Indian Restaurant          Women's Store            Donut Shop   
4     Fried Chicken Joint        Thai Restaurant    Athletics & Sports   
5              Playground           Dessert Shop    Dim Sum Restaurant   
6             Coffee Shop         Discount Store     Convenience Store   
7            Soccer Field          Metro Station           Bus Station   
8        Department Store     Dim Sum Restaurant                 Diner   
9            Skating Rink  General Entertainment         Women's Store   
10  Vietnamese Restaurant                Brewery    Chinese Restaurant   
11          Shopping Mall  Vietnamese Restaurant        Breakfast Spot   
12           Skating Rink         Breakfast Spot             Drugstore   
13        Thai Restaurant               Pharmacy          Noodle House   
14          Women's Store       Doner Restauran

In [27]:
fre1=pd.value_counts(scarborough['1st Most Common Venue'])
fre1

Fast Food Restaurant         2
Middle Eastern Restaurant    1
Bakery                       1
Coffee Shop                  1
Gas Station                  1
Pizza Place                  1
Department Store             1
Latin American Restaurant    1
Park                         1
Spa                          1
Intersection                 1
College Stadium              1
American Restaurant          1
Bar                          1
Indian Restaurant            1
Name: 1st Most Common Venue, dtype: int64

In [28]:
fre2=pd.value_counts(scarborough['2nd Most Common Venue'])
fre2

Convenience Store     2
Korean Restaurant     1
Bus Station           1
Chinese Restaurant    1
Bakery                1
Women's Store         1
Motel                 1
Department Store      1
Playground            1
Lounge                1
Auto Garage           1
Electronics Store     1
Bus Line              1
Café                  1
Pet Store             1
Name: 2nd Most Common Venue, dtype: int64

In [29]:
etobicoke=toronto_merged.loc[toronto_merged['BOROUGH']=='Etobicoke']
etobicoke

POSTALCODE    BOROUGH                                       NEIGHBORHOOD  \
88         M8V  Etobicoke       Humber Bay Shores, Mimico South, New Toronto   
89         M8W  Etobicoke                             Alderwood, Long Branch   
90         M8X  Etobicoke      The Kingsway, Montgomery Road, Old Mill North   
91         M8Y  Etobicoke  Humber Bay, King's Mill Park, Kingsway Park So...   
92         M8Z  Etobicoke  Kingsway Park South West, Mimico NW, The Queen...   
94         M9B  Etobicoke  Cloverdale, Islington, Martin Grove, Princess ...   
95         M9C  Etobicoke  Bloordale Gardens, Eringate, Markland Wood, Ol...   
99         M9P  Etobicoke                                          Westmount   
100        M9R  Etobicoke  Kingsview Village, Martin Grove Gardens, Richv...   
101        M9V  Etobicoke  Albion Gardens, Beaumond Heights, Humbergate, ...   
102        M9W  Etobicoke                                          Northwest   

      Latitude  Longitude 1st Most Common Venue 2nd Most Common Venue  \
88   43.605647 -79.501321                  Café           Coffee Shop   
89   43.602414 -79.543484           Pizza Place                   Gym   
90   43.653654 -79.506944                 River         Deli / Bodega   
91   43.636258 -79.498509        Breakfast Spot        Baseball Field   
92   43.628841 -79.520999                   Gym                Bakery   
94   43.650943 -79.554724                   NaN                   NaN   
95   43.643515 -79.577201                  Café           Coffee Shop   
99   43.696319 -79.532242           Pizza Place           Coffee Shop   
100  43.688905 -79.554724     Mobile Phone Shop              Bus Line   
101  43.739416 -79.588437         Grocery Store           Coffee Shop   
102  43.706748 -79.594054   Rental Car Location             Drugstore   

         3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
88                         Gym  Fast Food Restaurant            Restaurant   
89                 Coffee Shop              Pharmacy          Skating Rink   
90        Ethiopian Restaurant   Empanada Restaurant     Electronics Store   
91               Women's Store             Drugstore                 Diner   
92           Convenience Store        Discount Store         Burrito Place   
94                         NaN                   NaN                   NaN   
95              Cosmetics Shop            Beer Store     Convenience Store   
99   Middle Eastern Restaurant        Sandwich Place        Discount Store   
100                Pizza Place        Sandwich Place               Dog Run   
101                Pizza Place   Fried Chicken Joint        Sandwich Place   
102              Women's Store      Doner Restaurant          Dessert Shop   

           6th Most Common Venue 7th Most Common Venue 8th Most Common Venue  \
88           Fried Chicken Joint    Mexican Restaurant                Bakery   
89            Athletics & Sports                   Pub        Sandwich Place   
90   Eastern European Restaurant   Dumpling Restaurant             Drugstore   
91                Discount Store               Dog Run      Doner Restaurant   
92                  Burger Joint        Sandwich Place           Flower Shop   
94                           NaN                   NaN                   NaN   
95                   Pizza Place          Liquor Store   Empanada Restaurant   
99                  Intersection    Chinese Restaurant         Women's Store   
100                 Dessert Shop    Dim Sum Restaurant                 Diner   
101                   Beer Store  Fast Food Restaurant           Video Store   
102           Dim Sum Restaurant                 Diner        Discount Store   

    9th Most Common Venue       10th Most Common Venue  
88            Pizza Place                     Pharmacy  
89         Discount Store                 Dessert Shop  
90             Donut Shop             Doner Restaurant  
91             Donut Shop          Dumpl

In [30]:
fre1=pd.value_counts(etobicoke['1st Most Common Venue'])
fre1

Café                   2
Pizza Place            2
Grocery Store          1
Rental Car Location    1
Breakfast Spot         1
Mobile Phone Shop      1
River                  1
Gym                    1
Name: 1st Most Common Venue, dtype: int64

In [31]:
fre2=pd.value_counts(etobicoke['2nd Most Common Venue'])
fre2

Coffee Shop       4
Bus Line          1
Bakery            1
Drugstore         1
Baseball Field    1
Deli / Bodega     1
Gym               1
Name: 2nd Most Common Venue, dtype: int64

In [32]:
central_toronto=toronto_merged.loc[toronto_merged['BOROUGH']=='Central Toronto']
central_toronto

POSTALCODE          BOROUGH  \
44        M4N  Central Toronto   
45        M4P  Central Toronto   
46        M4R  Central Toronto   
47        M4S  Central Toronto   
48        M4T  Central Toronto   
49        M4V  Central Toronto   
63        M5N  Central Toronto   
64        M5P  Central Toronto   
65        M5R  Central Toronto   

                                         NEIGHBORHOOD   Latitude  Longitude  \
44                                      Lawrence Park  43.728020 -79.388790   
45                                   Davisville North  43.712751 -79.390197   
46                                 North Toronto West  43.715383 -79.405678   
47                                         Davisville  43.704324 -79.388790   
48                        Moore Park, Summerhill East  43.689574 -79.383160   
49  Deer Park, Forest Hill SE, Rathnelly, South Hi...  43.686412 -79.400049   
63                                           Roselawn  43.711695 -79.416936   
64                Forest Hill North, Forest Hill West  43.696948 -79.411307   
65                The Annex, North Midtown, Yorkville  43.672710 -79.405678   

   1st Most Common Venue 2nd Most Common Venue 3rd Most Common Venue  \
44                  Park              Bus Line           Swim School   
45                   Gym                 Hotel      Department Store   
46           Coffee Shop   Sporting Goods Shop           Yoga Studio   
47          Dessert Shop        Sandwich Place    Italian Restaurant   
48                   Gym            Playground          Tennis Court   
49                   Pub           Coffee Shop      Sushi Restaurant   
63                Garden         Women's Store      Doner Restaurant   
64         Jewelry Store    Mexican Restaurant      Sushi Restaurant   
65                  Café        Sandwich Place           Coffee Shop   

   4th Most Common Venue      5th Most Common Venue 6th Most Common Venue  \
44         Women's Store           Doner Restaurant    Dim Sum Restaurant   
45        Sandwich Place             Breakfast Spot     Food & Drink Shop   
46    Mexican Restaurant        Rental Car Location          Dessert Shop   
47                  Café                Pizza Place           Coffee Shop   
48       College Stadium              Deli / Bodega   Empanada Restaurant   
49    Light Rail Station      Vietnamese Restaurant            Restaurant   
63          Dessert Shop         Dim Sum Restaurant                 Diner   
64                 Trail              Women's Store               Dog Run   
65           Pizza Place  Middle Eastern Restaurant           Flower Shop   

   7th Most Common Venue        8th Most Common Venue 9th Most Common Venue  \
44                 Diner               Discount Store               Dog Run   
45                  Park        General Entertainment     Electronics Store   
46    Salon / Barbershop                         Café                 Diner   
47      Sushi Restaurant                          Gym                 Diner   
48     Electronics Store  Eastern European Restaurant   Dumpling Restaurant   
49          Liquor Store                  Supermarket           Pizza Place   
63        Discount Store                      Dog Run            Donut Shop   
64    Dim Sum Restaurant                        Diner        Discount Store   
65                   Pub            Indian Restaurant             BBQ Joint   

   10th Most Common Venue  
44             Donut Shop  
45              Gift Shop  
46   Fast Food Restaurant  
47         Farmers Market  
48              Drugstore  
49    American Restaurant  
63          Deli / Bodega  
64       Doner Restaurant  
65         History Museum

In [33]:
fre1=pd.value_counts(central_toronto['1st Most Common Venue'])
fre1

Gym              2
Café             1
Pub              1
Garden           1
Park             1
Coffee Shop      1
Dessert Shop     1
Jewelry Store    1
Name: 1st Most Common Venue, dtype: int64

In [34]:
fre2=pd.value_counts(central_toronto['2nd Most Common Venue'])
fre2

Sandwich Place         2
Mexican Restaurant     1
Women's Store          1
Bus Line               1
Coffee Shop            1
Sporting Goods Shop    1
Hotel                  1
Playground             1
Name: 2nd Most Common Venue, dtype: int64

In [35]:
west_toronto=toronto_merged.loc[toronto_merged['BOROUGH']=='West Toronto']
west_toronto

POSTALCODE       BOROUGH                                  NEIGHBORHOOD  \
76        M6H  West Toronto                  Dovercourt Village, Dufferin   
77        M6J  West Toronto                      Little Portugal, Trinity   
78        M6K  West Toronto  Brockton, Exhibition Place, Parkdale Village   
82        M6P  West Toronto                 High Park, The Junction South   
83        M6R  West Toronto                        Parkdale, Roncesvalles   
84        M6S  West Toronto                            Runnymede, Swansea   

     Latitude  Longitude 1st Most Common Venue 2nd Most Common Venue  \
76  43.669005 -79.442259              Pharmacy                Bakery   
77  43.647927 -79.419750                   Bar           Coffee Shop   
78  43.636847 -79.428191           Coffee Shop        Breakfast Spot   
82  43.661608 -79.464763                  Café                   Bar   
83  43.648960 -79.456325             Gift Shop    Italian Restaurant   
84  43.651571 -79.484450           Pizza Place                  Café   

   3rd Most Common Venue 4th Most Common Venue  5th Most Common Venue  \
76                   Bar                  Park          Grocery Store   
77           Men's Store            Restaurant  Vietnamese Restaurant   
78                  Café                Bakery                    Gym   
82    Mexican Restaurant       Thai Restaurant     Italian Restaurant   
83      Cuban Restaurant            Restaurant              Bookstore   
84           Coffee Shop              Tea Room     Italian Restaurant   

   6th Most Common Venue 7th Most Common Venue 8th Most Common Venue  \
76                  Bank  Gym / Fitness Center               Brewery   
77           Pizza Place                  Café      Asian Restaurant   
78               Stadium         Burrito Place            Restaurant   
82           Flea Market                Bakery                 Diner   
83                   Bar               Dog Run         Movie Theater   
84      Sushi Restaurant                 Diner                   Bar   

        9th Most Common Venue       10th Most Common Venue  
76  Middle Eastern Restaurant                  Music Venue  
77                Yoga Studio                    Juice Bar  
78               Climbing Gym        Performing Arts Venue  
82        Fried Chicken Joint                    Speakeasy  
83               Dessert Shop  Eastern European Restaurant  
84               Dessert Shop            Fish & Chips Shop

In [36]:
fre1=pd.value_counts(west_toronto['1st Most Common Venue'])
fre1

Coffee Shop    1
Gift Shop      1
Café           1
Bar            1
Pharmacy       1
Pizza Place    1
Name: 1st Most Common Venue, dtype: int64

In [37]:
fre2=pd.value_counts(west_toronto['2nd Most Common Venue'])
fre2

Italian Restaurant    1
Coffee Shop           1
Café                  1
Breakfast Spot        1
Bar                   1
Bakery                1
Name: 2nd Most Common Venue, dtype: int64

In [38]:
york=toronto_merged.loc[toronto_merged['BOROUGH']=='York']
york

POSTALCODE BOROUGH                                    NEIGHBORHOOD  \
73        M6C    York                              Humewood-Cedarvale   
74        M6E    York                             Caledonia-Fairbanks   
80        M6M    York  Del Ray, Keelesdale, Mount Dennis, Silverthorn   
81        M6N    York                   The Junction North, Runnymede   
98        M9N    York                                          Weston   

     Latitude  Longitude 1st Most Common Venue 2nd Most Common Venue  \
73  43.693781 -79.428191                 Trail                  Park   
74  43.689026 -79.453512                  Park         Women's Store   
80  43.691116 -79.476013    Turkish Restaurant           Coffee Shop   
81  43.673185 -79.487262               Brewery           Pizza Place   
98  43.706876 -79.518188                  Park     Convenience Store   

   3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
73                 Field          Hockey Arena         Women's Store   
74  Fast Food Restaurant                Market            Donut Shop   
80          Skating Rink                   Bar            Restaurant   
81        Breakfast Spot     Convenience Store             Drugstore   
98         Women's Store    Dim Sum Restaurant                 Diner   

   6th Most Common Venue 7th Most Common Venue 8th Most Common Venue  \
73          Dessert Shop    Dim Sum Restaurant                 Diner   
74    Dim Sum Restaurant                 Diner        Discount Store   
80        Sandwich Place               Dog Run          Dessert Shop   
81                 Diner        Discount Store               Dog Run   
98        Discount Store               Dog Run      Doner Restaurant   

   9th Most Common Venue 10th Most Common Venue  
73        Discount Store                Dog Run  
74               Dog Run       Doner Restaurant  
80    Dim Sum Restaurant                  Diner  
81      Doner Restaurant             Donut Shop  
98            Donut Shop              Drugstore

In [39]:
fre1=pd.value_counts(york['1st Most Common Venue'])
fre1

Park                  2
Trail                 1
Turkish Restaurant    1
Brewery               1
Name: 1st Most Common Venue, dtype: int64

In [40]:
fre2=pd.value_counts(york['2nd Most Common Venue'])
fre2

Coffee Shop          1
Women's Store        1
Pizza Place          1
Park                 1
Convenience Store    1
Name: 2nd Most Common Venue, dtype: int64

In [41]:
east_york=toronto_merged.loc[toronto_merged['BOROUGH']=='East York']
east_york

POSTALCODE    BOROUGH                     NEIGHBORHOOD   Latitude  \
35        M4B  East York  Woodbine Gardens, Parkview Hill  43.706397   
36        M4C  East York                 Woodbine Heights  43.695344   
38        M4G  East York                          Leaside  43.709060   
39        M4H  East York                 Thorncliffe Park  43.705369   
40        M4J  East York                     East Toronto  43.685347   

    Longitude 1st Most Common Venue 2nd Most Common Venue  \
35 -79.309937           Pizza Place  Fast Food Restaurant   
36 -79.318389              Pharmacy           Video Store   
38 -79.363452           Coffee Shop   Sporting Goods Shop   
39 -79.349372     Indian Restaurant                   Gym   
40 -79.338106                  Park     Convenience Store   

     3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
35               Pet Store              Pharmacy  Gym / Fitness Center   
36          Cosmetics Shop           Curling Ice          Skating Rink   
38  Furniture / Home Store        Sandwich Place          Burger Joint   
39             Supermarket                  Bank          Burger Joint   
40           Women's Store    Dim Sum Restaurant                 Diner   

   6th Most Common Venue 7th Most Common Venue  8th Most Common Venue  \
35             Gastropub                  Bank               Bus Line   
36            Beer Store                  Park  General Entertainment   
38            Restaurant      Sushi Restaurant            Supermarket   
39           Coffee Shop  Fast Food Restaurant            Gas Station   
40        Discount Store               Dog Run       Doner Restaurant   

   9th Most Common Venue       10th Most Common Venue  
35    Athletics & Sports                 Intersection  
36             Gift Shop  Eastern European Restaurant  
38     Fish & Chips Shop                 Liquor Store  
39         Grocery Store         Gym / Fitness Center  
40            Donut Shop                    Drugstore

In [42]:
fre1=pd.value_counts(east_york['1st Most Common Venue'])
fre1

Coffee Shop          1
Pizza Place          1
Pharmacy             1
Park                 1
Indian Restaurant    1
Name: 1st Most Common Venue, dtype: int64

In [43]:
fre2=pd.value_counts(east_york['2nd Most Common Venue'])
fre2

Fast Food Restaurant    1
Sporting Goods Shop     1
Gym                     1
Video Store             1
Convenience Store       1
Name: 2nd Most Common Venue, dtype: int64

In [44]:
east_toronto=toronto_merged.loc[toronto_merged['BOROUGH']=='East Toronto']
east_toronto

POSTALCODE       BOROUGH  \
37        M4E  East Toronto   
41        M4K  East Toronto   
42        M4L  East Toronto   
43        M4M  East Toronto   
87        M7Y  East Toronto   

                                         NEIGHBORHOOD   Latitude  Longitude  \
37                                        The Beaches  43.676357 -79.293031   
41                       The Danforth West, Riverdale  43.679557 -79.352188   
42                     The Beaches West, India Bazaar  43.668999 -79.315572   
43                                    Studio District  43.659526 -79.340923   
87  Business Reply Mail Processing Centre 969 Eastern  43.662744 -79.321558   

   1st Most Common Venue 2nd Most Common Venue 3rd Most Common Venue  \
37                  Park                 Trail                   Pub   
41      Greek Restaurant           Coffee Shop    Italian Restaurant   
42                  Park             Pet Store            Board Shop   
43                  Café           Coffee Shop                Bakery   
87           Yoga Studio         Auto Workshop            Smoke Shop   

     4th Most Common Venue        5th Most Common Venue 6th Most Common Venue  \
37       Health Food Store  Eastern European Restaurant   Dumpling Restaurant   
41  Furniture / Home Store               Ice Cream Shop           Yoga Studio   
42                 Brewery               Sandwich Place          Burger Joint   
43                 Brewery                    Gastropub    Italian Restaurant   
87                 Brewery                          Spa        Farmers Market   

   7th Most Common Venue 8th Most Common Venue 9th Most Common Venue  \
37             Drugstore            Donut Shop      Doner Restaurant   
41                 Diner  Caribbean Restaurant          Dessert Shop   
42         Burrito Place                   Pub                   Gym   
43   American Restaurant           Yoga Studio                   Bar   
87  Fast Food Restaurant         Burrito Place            Restaurant   

   10th Most Common Venue  
37           Dance Studio  
41                    Pub  
42       Sushi Restaurant  
43            Fish Market  
87       Recording Studio

In [45]:
fre1=pd.value_counts(east_toronto['1st Most Common Venue'])
fre1

Park                2
Yoga Studio         1
Greek Restaurant    1
Café                1
Name: 1st Most Common Venue, dtype: int64

In [46]:
fre2=pd.value_counts(east_toronto['2nd Most Common Venue'])
fre2

Coffee Shop      2
Auto Workshop    1
Trail            1
Pet Store        1
Name: 2nd Most Common Venue, dtype: int64

In [47]:
mississauga=toronto_merged.loc[toronto_merged['BOROUGH']=='Mississauga']
mississauga

POSTALCODE      BOROUGH                           NEIGHBORHOOD   Latitude  \
86        M7R  Mississauga  Canada Post Gateway Processing Centre  43.636966   

    Longitude 1st Most Common Venue 2nd Most Common Venue  \
86 -79.615819           Coffee Shop                 Hotel   

   3rd Most Common Venue     4th Most Common Venue 5th Most Common Venue  \
86                   Gym  Mediterranean Restaurant   Fried Chicken Joint   

        6th Most Common Venue 7th Most Common Venue 8th Most Common Venue  \
86  Middle Eastern Restaurant         Burrito Place        Sandwich Place   

   9th Most Common Venue 10th Most Common Venue  
86   American Restaurant             Donut Shop

In [48]:
queens_park=toronto_merged.loc[toronto_merged['BOROUGH']=="Queen's Park"]
queens_park

POSTALCODE       BOROUGH  NEIGHBORHOOD   Latitude  Longitude  \
93        M9A  Queen's Park  Queen's Park  43.667856 -79.532242   

   1st Most Common Venue 2nd Most Common Venue 3rd Most Common Venue  \
93           Coffee Shop                   Gym                  Park   

   4th Most Common Venue 5th Most Common Venue  6th Most Common Venue  \
93  Fast Food Restaurant           Salad Place  Portuguese Restaurant   

   7th Most Common Venue 8th Most Common Venue 9th Most Common Venue  \
93             Nightclub           Music Venue    Mexican Restaurant   

   10th Most Common Venue  
93              Juice Bar